content/docs/06.enterprise/05.instance/versioned-plugins.md

```diff
@@ -0,0 +1,100 @@
+        autoReloadEnabled: true
+        autoReloadInterval: 60s
+        defaultVersion: LATEST 
+```
+
+With remote storage enabled, installed plugins are stored in a plugins repository in the `_plugins/repository` path. For example, the below paths show the storage for 0.19.0 and 0.20.0 versions of the Shell script plugin:
+
+```bash
+_plugins/repository/io_kestra_plugin__plugin-script-shell__0_19_0
+_plugins/repository/io_kestra_plugin__plugin-script-shell__0_19_0.jar
+_plugins/repository/io_kestra_plugin__plugin-script-shell__0_20_0
+_plugins/repository/io_kestra_plugin__plugin-script-shell__0_20_0.jar
+```
+
+Artifact files are renamed with format: <groupId>__<artifactId>__<version> to be easily parseable (dots `.` are replaced to `_` for groupId and version).
+
+For locally stored plugins, configured by the `localRepositoryPath` attribute, the file path looks like `/tmp/kestra/plugins-repository`. For example, the following plugins are stored locally where the local repository contains a json `plugins.meta` file that contains meta data about remote plugins. This file is used for the synchronization where only plugins with detected changes are synchronized.
+
+```bash
+├── io_kestra_plugin__plugin-kafka__0_20_0.jar
+├── io_kestra_plugin__plugin-script-shell__0_20_0.jar
+├── io_kestra_plugin__plugin-terraform__0_20_0.jar
+├── io_kestra_plugin__plugin-transform-grok__0_20_0.jar
+└── plugins.meta
+```
+
+## Install versioned plugins
+
+Versioned plugins can be installed from the Kestra UI as well as programmatically.
+
+### From the UI
+
+Both Kestra official plugins and custom plugins can be installed from the UI. Navigate to the **Administration > Instance** section and then **Versioned Plugins**. From here, you can click **+ Install** and open up the full library of available plugins.
+
+![versioned-plugins-1](/docs/enterprise/versioned-plugins/versioned-plugins-1.png)
+
+From the list, search and select the plugin to install and select the version.
+
+![versioned-plugins-2](/docs/enterprise/versioned-plugins/versioned-plugins-2.png)
+
+After installing some plugins, your full list of versioned plugins are displayed. Kestra alerts you that a newer version of your plugin is available and gives you the capability to upgrade singular plugins as needed rather than all plugins to their newest version.
+
+![versioned-plugins-3](/docs/enterprise/versioned-plugins/versioned-plugins-3.png)
+
+For a custom plugin, after clicking **+ Install**, switch from Official plugin to Custom plugin. You need to specify two identifiers for each custom plugin installation:
+
+- Group ID: The group identifier of the plugin to be installed.
+- Artifact ID: The artifact identifier of the plugin to be installed.
+
+![versioned-plugins-5](/docs/enterprise/versioned-plugins/versioned-plugins-4.png)
+
+Additionally, instead of installing a new plugin, you can select **Upload** and upload a Plugin Archive File by choosing a stored valid Java Archive File (.jar).
```

Comment:
```suggestion
Additionally, instead of installing a new plugin, you can **Upload** a plugin by choosing a valid Java archive file (`.jar`).
```
cut not needed uppercase + shorter